src/services/GooglePlacesService.ts

```diff
@@ -87,7 +87,14 @@ class GooglePlacesService {
         },
       })
       .then((response: AxiosResponse) => {
-        const restaurantResults = response?.data?.result;
+        const restaurantData = response?.data;
+        const restaurantResults = restaurantData?.result;
+
+        // check if placeID is valid
+        if (restaurantData.status && restaurantData.status === 'NOT_FOUND') {
+          return this.refreshPlaceIDAndGetDetails(placeID);
+        }
```

Comment:
Are you referring to `response: AxiosResponse` ?

Reply:
ya

Reply:
Are you referring to the `?` in `const restaurantData = response?.data;`

We can't check for `response.status === 400` because it explicitly returns a `NOT_FOUND` status code when the place ID is invalid which is the error we want to handle for this PR

I couldn't find a specific list of status codes for Google Places API- this one is for Android vers but here are some [examples](https://developers.google.com/places/android-sdk/reference/com/google/android/libraries/places/api/net/PlacesStatusCodes)

Reply:
Ohh I see, I thought the `NOT_FOUND` was 400 status code. 

But my other question is, why do we need `response?.data`, could it be `response.data`

Reply:
Hmm.. I guess it could be- but I wanted to be safe! I couldn't find any particular scenarios where past users were returned `undefined` for Places API but it's happened for Maps API and Nearby Search, etc so I think it's better safe than sorry unless it's harmful to include the extra check?

Reply:
Maps API and Nearby Search returned undefined responses from axios? Isn't that very unlikely because the signature for `axios.get` always returns the interface `AxiosResponse`?